pruner/graph.go

```diff
@@ -0,0 +1,328 @@
+package pruner
+
+import (
+	"bufio"
+	"container/list"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/registry/storage"
+	"github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+)
+
+const (
+	confirmText = "Do you really want to permanently delete objects? [y/n]: "
+)
+
+type byName []string
```

Comment:
You can use `sort.StringSlice` instead, which is exactly the same implementation as this